src/test/scala/com/databricks/spark/redshift/RedshiftSourceSuite.scala

```diff
@@ -0,0 +1,486 @@
+        TestUtils.toTimestamp(2015, 6, 3, 0, 0, 0)),
+      Row(0.toByte, false, null, -1234152.123124981, 100000.0f, null, 1239012341823719L, 24, "___|_123", null),
+      Row(List.fill(10)(null): _*))
+
+
+  /**
+   * Spark Context with hadoop file overridden to point at our local test data file for this suite,
+   * no-matter what temp directory was generated and requested.
+   */
+  private var sc: SparkContext = _
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    sc = new TestContext
+  }
+
+  override def afterAll(): Unit = {
+    val temp = new File(tempDir)
+    val tempFiles = temp.listFiles()
+    if(tempFiles != null) tempFiles foreach {
+      case f => if(f != null) f.delete()
+    }
+    temp.delete()
+
+    sc.stop()
+    super.afterAll()
+  }
+
+  /**
+   * Set up a mocked JDBCWrapper instance that expects a sequence of queries matching the given
+   * regular expressions will be executed, and that the connection returned will be closed.
+   */
+  def mockJdbcWrapper(expectedUrl: String, expectedQueries: Seq[Regex]): JDBCWrapper = {
+    val jdbcWrapper = mock[JDBCWrapper]
+    val mockedConnection = mock[Connection]
+
+    (jdbcWrapper.getConnector _).expects(*, expectedUrl, *).returning(() => mockedConnection)
+
+    inSequence {
+      expectedQueries foreach { r =>
+        val mockedStatement = mock[PreparedStatement]
+        (mockedConnection.prepareStatement(_: String))
+          .expects(where {(sql: String) => r.findFirstMatchIn(sql).nonEmpty})
+          .returning(mockedStatement)
+        (mockedStatement.execute _).expects().returning(true)
+      }
+
+      (mockedConnection.close _).expects()
+    }
+
+    jdbcWrapper
+  }
+
+  /**
+   * Prepare the JDBC wrapper for an UNLOAD test.
+   */
+  def prepareUnloadTest(params: Map[String, String]) = {
+    val jdbcUrl = params("url")
+    val jdbcWrapper = mockJdbcWrapper(jdbcUrl, Seq("UNLOAD.*".r))
+
+    // We expect some extra calls to the JDBC wrapper,
+    // to register the driver and retrieve the schema.
+    (jdbcWrapper.registerDriver _)
+      .expects(*)
+      .anyNumberOfTimes()
+    (jdbcWrapper.resolveTable _)
+      .expects(jdbcUrl, "test_table", *)
+      .returning(TestUtils.testSchema)
+      .anyNumberOfTimes()
+
+    jdbcWrapper
+  }
+
+  test("DefaultSource can load Redshift UNLOAD output to a DataFrame") {
+
+    val params = Map("url" -> "jdbc:postgresql://foo/bar",
+      "tempdir" -> "tmp",
+      "dbtable" -> "test_table",
+      "aws_access_key_id" -> "test1",
+      "aws_secret_access_key" -> "test2")
+
+    val jdbcWrapper = prepareUnloadTest(params)
+    val testSqlContext = new SQLContext(sc)
+
+    // Assert that we've loaded and converted all data in the test file
+    val source = new DefaultSource(jdbcWrapper)
+    val relation = source.createRelation(testSqlContext, params)
+    val df = testSqlContext.baseRelationToDataFrame(relation)
+
+    df.rdd.collect() zip expectedData foreach {
```

Comment:
This is a dangerous way to compare answers since this will never fail if the query returns an empty result:

```scala
scala> (Seq.empty[Int] zip Seq(1, 2, 3)).foreach(_ => assert(false))
```

This will not throw any error.